packages/amplify-graphql-auth-transformer/src/graphql-auth-transformer.ts

```diff
@@ -113,26 +113,26 @@ import { AuthVTLGenerator } from './vtl-generator/vtl-generator';
  */
 const getReadRolesForField = (acm: AccessControlMatrix, readRoles: Array<string>, fieldName: string): Array<string> => {
   const hasCognitoPrivateRole =
-    readRoles.some((r) => r === 'userPools:private') &&
-    acm.isAllowed('userPools:private', fieldName, 'get') &&
-    acm.isAllowed('userPools:private', fieldName, 'list') &&
-    acm.isAllowed('userPools:private', fieldName, 'sync') &&
-    acm.isAllowed('userPools:private', fieldName, 'search') &&
-    acm.isAllowed('userPools:private', fieldName, 'listen');
+    readRoles.some((r) => r === `${acm.getName()}:userPools:private`) &&
```

Comment:
Here and throughout: can we precalc role names rather than reusing the same string interpolation multiple times?

Reply:
Updated. Pre-calculating and reusing the value whenever possible.